src/logger/logger.ts

```diff
@@ -1,12 +1,139 @@
-import pino from "pino"
+import _ from "lodash"
+import pino, { Logger as PinoLogger } from "pino"
+
+import { StackFrame, get, parse } from "./stack-trace"
+
+interface WarnMeta {
+  params: Record<string, unknown>
```

Comment:
Why do we need params in WarnMeta?

Reply:
my assumption is that when people call `warn`, it's dangerous behaviour that should be looked at still. hence, we want to capture as much context as possible (such as `params` of the calling function) to aid in debugging